contribute/engineering/backend/instrumentation.md

```diff
@@ -0,0 +1,207 @@
+
+It's also possible to configure multiple loggers:
+
+```ini
+[log]
+filters = rendering:debug \
+          ; alerting.notifier:debug \
+          oauth.generic_oauth:debug \
+          ; oauth.okta:debug \
+          ; tsdb.postgres:debug \
+          ; tsdb.mssql:debug \
+          ; provisioning.plugins:debug \
+          ; provisioning:debug \
+          ; provisioning.dashboard:debug \
+          ; provisioning.datasources:debug \
+          datasources:debug \
+          data-proxy-log:debug
+```
+
+## Metrics
+
+Metrics are quantifiable measurements that reflect the health and performance of applications or infrastructure.
+
+Consider using metrics to provide real-time insight into the state of resources. If you want to know how responsive your application is or identify anomalies that could be early signs of a performance issue, metrics are a key source of visibility.
+
+There are many possible types of metrics that can be tracked. One popular method for defining metrics is the [RED method](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/).
+
+### Metric types
+
+See [Prometheus metric types](https://prometheus.io/docs/concepts/metric_types/) for a list and description of the different metric types you can use and when to use them.
+
+### Naming conventions
+
+Use the namespace _grafana_ as that would prefix any defined metric names with `grafana_`. This will make it clear for operators that any metric named `grafana_*` belongs to Grafana.
+
+Use snake_case style when naming metrics, e.g. _http_request_duration_seconds_ instead of _httpRequestDurationSeconds_.
+
+Use snake_case style when naming labels, e.g. _status_code_ instead of _statusCode_.
+
+If metric type is a _counter_, name it with a `_total` suffix, e.g. _http_requests_total_.
+
+If metric type is a _histogram_ and you're measuring duration, name it with a `_<unit>` suffix, e.g. _http_request_duration_seconds_.
+
+If metric type is a _gauge_, name it to denote it's a value that can increase and decrease , e.g. _http_request_in_flight_.
+
+### Label values and high cardinality
+
+Be careful with what label values you add/accept. Using/allowing too many label values could result in [high cardinality problems](https://grafana.com/blog/2022/02/15/what-are-cardinality-spikes-and-why-do-they-matter/).
+
+If label values originates from user input they **should** be validated. Use `metricutil.SanitizeLabelName(<label value>`) from _pkg/infra/metrics/metricutil_ package to sanitize label names. Very **important** to only allow a pre-defined set of labels to minimize the risk of high cardinality problems.
+
+Be **careful** to not expose any sensitive information in label values, e.g. secrets, credentials etc.
```

Comment:
I guess the same applies to logs, where its super easy to accidentally put a whole object worth of info

Reply:
Yes, very good point :+1:

Reply:
Fixed